photoeditor/src/main/java/ja/burhanrashid52/photoeditor/PhotoEditorImpl.java

```diff
@@ -294,18 +294,24 @@ public void saveAsFile(@NonNull final String imagePath, @NonNull final OnSaveLis
     }
 
     @Override
-    @SuppressLint("StaticFieldLeak")
     public void saveAsFile(@NonNull final String imagePath,
                            @NonNull final SaveSettings saveSettings,
                            @NonNull final OnSaveListener onSaveListener) {
         Log.d(TAG, "Image Path: " + imagePath);
         parentView.saveFilter(new OnSaveBitmap() {
             @Override
             public void onBitmapReady(Bitmap saveBitmap) {
-                PhotoSaverTask photoSaverTask = new PhotoSaverTask(parentView, mBoxHelper);
-                photoSaverTask.setOnSaveListener(onSaveListener);
-                photoSaverTask.setSaveSettings(saveSettings);
-                photoSaverTask.execute(imagePath);
+                if (saveSettings.isSyncSave()) {
```

Comment:
Maybe this can be extracted in a dedicated method used both by saveAsFile and saveAsBitmap

Reply:
I didn't understand your suggestion. These are 2 different classes (couldn't reuse) and also have different methods (one without and one with argument, at the last command), I'd be happy to edit the PR if you could tell me how could this be improved.